sygnal/apnspushkin.py

```diff
@@ -422,25 +422,40 @@ def _get_payload_full(self, n, log):
                 badge = 0
             badge += n.counts.missed_calls
 
-        if badge is not None:
-            aps["badge"] = badge
-
-        if loc_key:
-            aps["mutable-content"] = 1
-
         if loc_key is None and badge is None:
             log.info("Nothing to do for alert of type %s", n.type)
             return None
 
         payload = {}
 
+        if (
+            n.type is not None
+            and device.data is not None
+            and device.data["default_payload"] is not None
+        ):
+            payload = device.data["default_payload"]
+
+        if "aps" not in payload:
+            payload["aps"] = {}
```

Comment:
you can write this more succinctly:

```suggestion
        payload.setdefault("aps", {})
```